### ISSUE TITLE
sweetalert2_1.default is not a function

### ISSUE THREAD

**Issue by readme42** [limonte/sweetalert2#390]
I'm trying to switch from sweetalert to sweetalert2. 

Is anyone using this successfully with webpack2 and angular2?

Whatever i try, i always get this error: 

`Error in :0:0 caused by: sweetalert2_1.default is not a function`

My component is the most simple i can guess

```
import swal from "sweetalert2";
import {Component, OnInit} from "@angular/core";

@Component({
    selector: "dashboard",
    template: require("./dashboard.template.html")
})
export class Dashboard implements OnInit {
    ngOnInit() {
        swal("Hi from webpack!");
    }
}
```

The code generated by webpack looks like this:

```
var sweetalert2_1 = __webpack_require__(1334);
var core_1 = __webpack_require__(0);
var Dashboard = (function () {
    function Dashboard() {
    }
    Dashboard.prototype.ngOnInit = function () {
        sweetalert2_1.default("Hi from webpack!");
    };
    Dashboard = __decorate([
        core_1.Component({
            selector: "dashboard",
            template: __webpack_require__(1387)
        }), 
        __metadata('design:paramtypes', [])
    ], Dashboard);
    return Dashboard;
}());
exports.Dashboard = Dashboard;
```

EDIT: 
**When replacing the import with require the bundle is fine and it works** 

So replace 
`import swal from "sweetalert2";`
with
`var swal = require("sweetalert2");`

But then, no TypeScript Type definitions are recognized :/

Does anyone know how to workaround?

**Comment by toverux** [comment]
@readme42 I presume you're using TypeScript 1.x?
Please try my test branch to fix this by running `npm i -S toverux/sweetalert2#fix/named-exports`, and use the import syntax `import swal from "sweetalert2";`.
That should work.

@limonte I think that's because older versions of TypeScript don't like implicit default exports. By changing the way we export the lib using named exports, that should work everywhere, and fix usage in TypeScript 1.x. TypeScript 2.x support implicit default exports for interoperability if I remember correctly. Before upgrading to TS 2 in one of my apps, I also had to use `import swal = require('sweetalert2');`, otherwise it would have resulted in the same error.

**Comment by carganilla** [comment]
I'm using TypeScript 2.1.4, webpack 1.14.0, Angular2 and had the same issue.  I tried the test branch and it works!

**Comment by limonte** [comment]
Thanks for the feedback @carganilla! @toverux go ahead and create the PR from the `fix/named-exports` branch :rocket:

**Comment by readme42** [comment]
@toverux 
I'm using TypeScript 2.0.10, not the 1.x branch. 

Im sorry, but i will not be able to test your test branch before 03.01.2017 - but then i will. Thanks for your fast response :)

**Comment by readme42** [comment]
@toverux (CC @limonte )
My promised feedback: your branch `fix/named-exports` works like a charm :)

**Comment by toverux** [comment]
Ok, I'm back in the game after a few days afk.
I understand why it breaks with Webpack 1.

As you may know, TypeScript does not come with a compatibility layer concerning `default` imports. For example, Babel will check if the imported value has `__esModule: true` in the imported value, and if so, take the `default` property. If the imported value or object haven't the __esModule marker, then the raw value will be taken as the `default`.

TypeScript doesn't do that. It hardcodes `.default` in the output JavaScript. That's why named exports work well with TypeScript in its default configuration.
Nothing changes in TypeScript 2. But, with Webpack 2, you may want to modify your `tsconfig.json` to add :

```json
{
  "compilerOptions": {
    "module": "es2015"
  }
}
```

With this, TS will still compile your source down to ES 5, but will let `import` / `export` declaration *untouched*. Then, Webpack 2 (that understands ES 2015 modules) will apply its own `default` wiring when compiling in ES 5, Webpack-specific format. No more `.default` but a lovely compatibility layer. That's why Sweetalert on Webpack 2 with the right tsconfig.json works out of the box.

**Comment by toverux** [comment]
See #391 for progress and https://github.com/toverux/sweetalert2-build-demos/tree/master/webpack1-typescript2 that I've juste created to test it (I will add more configurations later).

**Comment by limonte** [comment]
@readme42 Fixed in [v6.3.0](https://github.com/limonte/sweetalert2/releases/tag/v6.3.0)

All credits to @toverux :1st_place_medal:

**Comment by readme42** [comment]
great work, thanks a lot :)
